interfaces/udev/backend.go

```diff
@@ -74,37 +108,35 @@ func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repo
 //
 // If the method fails it should be re-tried (with a sensible strategy) by the caller.
 func (b *Backend) Remove(snapName string) error {
-	glob := fmt.Sprintf("70-%s.rules", interfaces.SecurityTagGlob(snapName))
-	return ensureDirState(dirs.SnapUdevRulesDir, glob, nil, snapName)
-}
-
-func ensureDirState(dir, glob string, content map[string]*osutil.FileState, snapName string) error {
-	var errReload error
-	changed, removed, errEnsure := osutil.EnsureDirState(dir, glob, content)
-	if len(changed) > 0 || len(removed) > 0 {
-		// Try reload the rules regardless of errEnsure.
-		errReload = ReloadRules()
+	rulesFilePath := snapRulesFilePath(snapName)
+	if _, err := os.Stat(rulesFilePath); os.IsNotExist(err) {
```

Comment:
Checking for existence before removal is unnecessary. Just remove and check the error there (`...; err != nil && !os.IsNotExist {`).